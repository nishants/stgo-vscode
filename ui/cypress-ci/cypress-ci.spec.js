import React from 'react';
import {mount} from 'enzyme';

import CypressBuildTab from './index';

const comletedBuild = {status: 2, _links: {web: {href: 'http://path-to-build-1'}}, sourceVersion: "123"};
const CancellingBuild = {status: 3, _links: {web: {href: 'http://path-to-build-1'}}, sourceVersion: "456"};
const buildInProgress = {status: 1, _links: {web: {href: 'http://path-to-build-2'}}, sourceVersion: "456"};

describe('ui/cypress-ci/cypress-ci', () => {
  it('should disable button if a build is in progress', () => {
    const getCypressBuilds = jest.fn();
    const triggerBuild = jest.fn();

    const wrapped = mount(<CypressBuildTab
      getCypressBuilds={getCypressBuilds}
      currentBranchName='zyx-branch'
      callBack={triggerBuild}
      data={[
        comletedBuild,
        buildInProgress
      ]}
    />);

    expect(wrapped).toMatchSnapshot();
  });

  it('should enable button if there are no builds in progress', () => {
    const getCypressBuilds = jest.fn();
    const triggerBuild = jest.fn();

    const wrapped = mount(<CypressBuildTab
      getCypressBuilds={getCypressBuilds}
      currentBranchName='zyx-branch'
      callBack={triggerBuild}
      data={[
        CancellingBuild,
        comletedBuild
      ]}
    />);

    expect(wrapped).toMatchSnapshot();
  });

});