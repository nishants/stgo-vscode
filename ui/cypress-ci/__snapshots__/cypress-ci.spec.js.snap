// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/cypress-ci/cypress-ci should disable button if a build is in progress 1`] = `
<CypressCi
  callBack={[MockFunction]}
  currentBranchName="zyx-branch"
  data={
    Array [
      Object {
        "_links": Object {
          "web": Object {
            "href": "http://path-to-build-1",
          },
        },
        "buildTime": "Mon Mar 23 2020 19:04:05
",
        "sourceVersion": "123",
        "status": 2,
      },
      Object {
        "_links": Object {
          "web": Object {
            "href": "http://path-to-build-2",
          },
        },
        "buildTime": "Mon Mar 23 2020 19:04:05
",
        "sourceVersion": "456",
        "status": 1,
      },
    ]
  }
  getCypressBuilds={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <div
    className="cypressCi"
  >
    <button
      disabled={true}
      onClick={[MockFunction]}
    >
      Run Cypress Tests
    </button>
    <CypressBuildInfo
      buildTime="Mon Mar 23 2020 19:04:05
"
      commitBuildHref="http://path-to-build-1"
      commitId="123"
      key="123"
      status={2}
    >
      <div
        className="build-info"
        key="123"
      >
        <div>
          Mon Mar 23 2020 19:04:05

        </div>
        <div>
          123
        </div>
        <span
          className="status-flag warning"
        >
          <a
            href="http://path-to-build-1"
          >
            Unknown
            <i
              className="icon fas fa-external-link-square-alt"
            />
          </a>
        </span>
      </div>
    </CypressBuildInfo>
    <CypressBuildInfo
      buildTime="Mon Mar 23 2020 19:04:05
"
      commitBuildHref="http://path-to-build-2"
      commitId="456"
      key="456"
      status={1}
    >
      <div
        className="build-info"
        key="456"
      >
        <div>
          Mon Mar 23 2020 19:04:05

        </div>
        <div>
          456
        </div>
        <span
          className="status-flag warning"
        >
          <a
            href="http://path-to-build-2"
          >
            Unknown
            <i
              className="icon fas fa-external-link-square-alt"
            />
          </a>
        </span>
      </div>
    </CypressBuildInfo>
  </div>
</CypressCi>
`;

exports[`ui/cypress-ci/cypress-ci should enable button if there are no builds in progress 1`] = `
<CypressCi
  callBack={[MockFunction]}
  currentBranchName="zyx-branch"
  data={
    Array [
      Object {
        "_links": Object {
          "web": Object {
            "href": "http://path-to-build-1",
          },
        },
        "buildTime": "Mon Mar 23 2020 19:04:05
",
        "sourceVersion": "456",
        "status": 3,
      },
      Object {
        "_links": Object {
          "web": Object {
            "href": "http://path-to-build-1",
          },
        },
        "buildTime": "Mon Mar 23 2020 19:04:05
",
        "sourceVersion": "123",
        "status": 2,
      },
    ]
  }
  getCypressBuilds={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <div
    className="cypressCi"
  >
    <button
      disabled={false}
      onClick={[MockFunction]}
    >
      Run Cypress Tests
    </button>
    <CypressBuildInfo
      buildTime="Mon Mar 23 2020 19:04:05
"
      commitBuildHref="http://path-to-build-1"
      commitId="456"
      key="456"
      status={3}
    >
      <div
        className="build-info"
        key="456"
      >
        <div>
          Mon Mar 23 2020 19:04:05

        </div>
        <div>
          456
        </div>
        <span
          className="status-flag warning"
        >
          <a
            href="http://path-to-build-1"
          >
            Unknown
            <i
              className="icon fas fa-external-link-square-alt"
            />
          </a>
        </span>
      </div>
    </CypressBuildInfo>
    <CypressBuildInfo
      buildTime="Mon Mar 23 2020 19:04:05
"
      commitBuildHref="http://path-to-build-1"
      commitId="123"
      key="123"
      status={2}
    >
      <div
        className="build-info"
        key="123"
      >
        <div>
          Mon Mar 23 2020 19:04:05

        </div>
        <div>
          123
        </div>
        <span
          className="status-flag warning"
        >
          <a
            href="http://path-to-build-1"
          >
            Unknown
            <i
              className="icon fas fa-external-link-square-alt"
            />
          </a>
        </span>
      </div>
    </CypressBuildInfo>
  </div>
</CypressCi>
`;
